### ISSUE TITLE
Problem fetching dependencies

### ISSUE THREAD

**Issue by raproth** [polynote/polynote#850]
I have a Notebook (spark 2.4.4) which needs some dependencies, these are available on our enterprise repository (artifactory). Therefore I set "https://bin.mycompany.com/artifactory/mvn/" as a custom Maven resolver in the Notebook config.

The spark jobs get stuck and don't finish. I inspected the executor logs and see that the executors try to reach  https://repo1.maven.org, which they can't (due to a firewall).

Inspecting the spark environment (in SparkUI), I see that several entries from `spark.jars` have URLs pointing to  https://repo1.maven.org (and some others point to https://bin.mycompany.com/artifactory/mvn/). 

How can I configure polynote to only use my enterprise repository? I'm not allowed to use https://repo1.maven.org

**Comment by JD557** [comment]
Can you check if my comments from #786 help?

(I wasn't able to reproduce that problem, I've been downloading dependencies from my enterprise repository just fine)

**Comment by raproth** [comment]
I don't have any credentials configured (no need to authenticate for the repo), I could still dry to add a file.
What I don't really get is how the dependency resolution works. Is coursier fetching dependencies for the driver code and spark-submit (-jars) for the  remote code? Or is there only 1 dependency resolution for "local" and "remote" machines?
If I start the notebook, I see some progress bars for the dependency fetching, unfortunately I don't see anything in the logs...

**Comment by jeremyrsmith** [comment]
@raproth Maybe you're being bitten by how dependencies are currently handed to Spark. We resolve them from the Polynote server, but then we just give their URLs to Spark for downloading. This probably isn't ideal for authenticated repos or for repos that can't be accessed from the executor (but can be accessed from the Polynote server).

The reason this was done is so that every executor doesn't have to fetch large dependencies from the single machine running the driver. But, this behavior definitely needs to be made configurable for cases like yours (and we should have more configurability in general around dependency resolution, like being able to disable Maven Central).

**Comment by raproth** [comment]
@jeremyrsmith  yes, this is/was probabely the issue. The driver was able to reach https://repo1.maven.org/, the executors not! This solves my issue for now, thanks.

Is there a way to add verbosity/logging to the dependency resolution? I don't see any logs concering coursier, neither in brower console nor in stdout from polynote.py?
